File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l=0
        result = 0

        for r in range(len(prices)):
            profit = 0
            if prices[r]>prices[l]:
                profit = prices[r]-prices[l]
                result = max(result,profit)
            else:
                l=r
        return result
        

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.java
class Solution {
    public int characterReplacement(String s, int k) {
        Map<Character,Integer> map = new HashMap<>();
        int maxf=0;
        int result=0;
        int l=0;

        for(int r=0; r<s.length(); r++){
            char cr = s.charAt(r);
            char cl = s.charAt(l);
            map.put(cr,map.getOrDefault(cr,0)+1);
            maxf = Math.max(maxf,map.get(cr));
            while((r-l+1) - maxf > k){
                map.put(cl,map.get(cl)-1);
                l++;
            }
            result = Math.max((r-l+1),result);
        }
        return result;

    }
}
